test/Esprima.Tests.SourceGenerators/Snapshots/StringMatcherGeneratorTests.ScannerStringMatchingGeneration.01StringMatchers.g.verified.cs

```diff
@@ -228,158 +253,431 @@ public partial class Scanner
 
     private static partial string? TryGetInternedTwoCharacterPunctuator(ReadOnlySpan<char> input)
     {
-        switch (input)
-        {
-            case "&&":
-                return "&&";
-            case "||":
-                return "||";
-            case "==":
-                return "==";
-            case "!=":
-                return "!=";
-            case "+=":
-                return "+=";
-            case "-=":
-                return "-=";
-            case "*=":
-                return "*=";
-            case "/=":
-                return "/=";
-            case "++":
-                return "++";
-            case "--":
-                return "--";
-            case "<<":
-                return "<<";
-            case ">>":
-                return ">>";
-            case "&=":
-                return "&=";
-            case "|=":
-                return "|=";
-            case "^=":
-                return "^=";
-            case "%=":
-                return "%=";
-            case "<=":
-                return "<=";
-            case ">=":
-                return ">=";
-            case "=>":
-                return "=>";
-            case "**":
-                return "**";
-            default:
+            if (input[0] == '&')
```

Comment:
nit 2: using `switch` instead of `if`s (at least for the first char) would make it a bit clearer to the human eye. Also, in some cases [`switch` may result in more performant code](https://stackoverflow.com/a/767839/8656352).

Reply:
i'ts getting a bit nasty to handle nested constructs, I'll leave this as future improvement...